### PR TITLE
Checksum support

### PR DESCRIPTION
**Pull request by tsvisabo**

### CODE REVIEW

File: common/helpers/tests/test_checksum.cpp [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+      "EQXJHD9VY";
+
+  const std::string CHECKSUM =
+      "CFOHVYSWYAKMFOPCSUXAYJQFXIZCAFLJSLZQNNJMYCBEERFGZ9C9GYDZYDADZIHJMUWGIYRP"
+      "XOEDEOMRC";
+
+  char* out = iota_checksum(ADDRESS.c_str(), ADDRESS.size(), 9);
+
+  EXPECT_EQ(out, CHECKSUM.substr(CHECKSUM.size() - 9, 9));
+
```
Comment (th0br0) [inline common/helpers/tests/test_checksum.cpp]: could you fix the newlines here?

File: common/kerl/hash.c [hunk 2]
```diff
@@ -0,0 +1,11 @@
+#include "hash.h"
+#include "common/trinary/tryte.h"
+
+void kerl_hash(const trit_t* const trits, size_t len, trit_t* out, Kerl* kerl){
+char digest[HASH_LENGTH];
```
Comment (th0br0) [inline common/kerl/hash.c]: formatting

**Review by th0br0 — CHANGES_REQUESTED** [review]

File: common/helpers/checksum.c [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+#include <stdlib.h>
+#include <string.h>
+
+#include "checksum.h"
+#include "common/kerl/hash.h"
+#include "common/sign/v1/iss_kerl.h"
+#include "common/trinary/trit_tryte.h"
+
+
+#define RADIX 3
```
Comment (th0br0) [inline common/helpers/checksum.c]: don't we have this `#define`d elsewhere already?

File: common/helpers/checksum.c [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+
+char* iota_checksum(const char* input, const size_t inputLength,
+                    const size_t checksumLength) {
+  Kerl kerl;
+  init_kerl(&kerl);
+
+  char inputTrits[sizeof(char) * inputLength * RADIX];
+  trit_t trits_hash[HASH_LENGTH];
+  char hash[HASH_LENGTH / RADIX];
```
Comment (th0br0) [inline common/helpers/checksum.c]: not needed

File: common/helpers/checksum.c [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+  char* checksumTrytes;
+
+  trytes_to_trits((tryte_t*)input, inputTrits, inputLength);
+  kerl_hash(inputTrits, inputLength * RADIX, trits_hash, &kerl);
+  trits_to_trytes((trit_t*)trits_hash, (tryte_t*)hash, HASH_LENGTH);
```
Comment (th0br0) [inline common/helpers/checksum.c]: move this after `checksumTrytes` is alloced so you can just write the trytes to there

File: common/helpers/checksum.c [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+
+  checksumTrytes = calloc(1 + checksumLength, sizeof(tryte_t));
+
+  memcpy(checksumTrytes, &hash[HASH_LENGTH / RADIX - checksumLength],
```
Comment (th0br0) [inline common/helpers/checksum.c]: not needed with ^

File: common/helpers/checksum.c [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+         checksumLength);
+  checksumTrytes[checksumLength] = '\0';
```
Comment (th0br0) [inline common/helpers/checksum.c]: `calloc` already initialises to 0

**Review by th0br0 — CHANGES_REQUESTED** [review]

File: common/helpers/checksum.c (hunk 3 above)
Comment (tsvisabo) [reply common/helpers/checksum.c]: move to trit_tryte.h

File: common/helpers/checksum.c (hunk 4 above)
Comment (tsvisabo) [reply common/helpers/checksum.c]: removed

**Review by th0br0 — APPROVED** [review]

**Pull request merged**